rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,996 @@
+
+**Bundle size:** ~12kb
+
+ReactAria’s behavior is very different from fluentui, it’s FocusScope doesn’t provide roving index behavior instead it’s like a focus trap. Which is equivalent to FluentUI’s FocusTrapZone
+
+Although ReactAria does provides low level primitives to manage focus like `useFocusManager`
+
+**Pros:**
+
+- Provides other helpful features with it’s focus package, which makes it worth the bundle size compared to other solutions, with other libs we will have to look for alternatives for:
+  - [FocusScope](https://react-spectrum.adobe.com/react-aria/FocusRing.html) (Trapping focus)
+  - [FocusRing](https://react-spectrum.adobe.com/react-aria/FocusRing.html)
+  - [useFocusRing](https://react-spectrum.adobe.com/react-aria/useFocusRing.html)
+- Provides low level primitives for building roving index like behaviors, eg: useFocusManager, getFocusableTreeWalker, createFocusManager
+- Picking react aria will also encourage use of react-native-aria & the eco system of react aria.
+
+**Cons:**
+
+- Doesn’t provide roving index behavior by default
+- A lot of manual logic and code has to be written to implement a proper Roving index component which handles edge cases,
+- The implementation in the codesandbox has various bugs & edge cases, for eg: initially the elements don't have tabIndex set, these kinds of edge cases we need to solve for.
+
+</details>
+
+**Ariakit** - [Demo](https://codesandbox.io/s/blade-comp-ariakit-composite-ewc9i4?file=/src/App.tsx) | [Doc](https://reakit.io/docs/composite/)
+
+<details>
+  <summary>Pros and cons</summary>
+
+Ariakit is the only out of the three which properly implements the composite widget specification.
+It even supports virtual focus with aria-activedescendant. It’s the most complete implementation.
+
+**Bundle size:** ~15kb tree shakable depending on which features we use
+
+**Pros:**
+
+- Easy to use
+- Most spec compliant and holistic implementation
+- Very flexible with its hooks based architecture. Composition is ariakit’s best selling point
+- Lots of other features, like [composite groups, typeahead, input, hover, separator](https://github.com/ariakit/ariakit/tree/main/packages/ariakit/src/composite)
+- [Supports 2d navigation](https://reakit.io/docs/composite/#two-dimensional-navigation) same as FluentUI
+- Support virtual focus + roving index. ++respect!
+- Ariakit also provides primitives and utilities like useFocusTrap & useFocusable, although they will increase the bundle size further
+
+**Cons:**
+
+- The biggest issue with ariakit is that it’s not stable yet :(  
+  So ariakit is the successor to reakit, Haz has been developing it for over 4 years now.
+  But as the doc says ariakit is better but still the API is not stable and can have breaking changes
+- The new version of ariakit lacks documentation
+- No separate package scope for their focus utilities, we will have to rely on treeshaking to be able to shave all other unnecessary components.
+
+</details>
+
```

Comment:
Which one are we going ahead with? Needs a conclusion section here. Or if we've open question on this.